pkg/cloudmap/resource_manager.go

```diff
@@ -8,437 +8,355 @@ import (
 	Cleanup(ctx context.Context, vn *appmesh.VirtualNode) error
 }
 
-func NewCloudMapResourceManager(
+func NewDefaultResourceManager(
 	k8sClient client.Client,
 	cloudMapSDK services.CloudMap,
 	virtualNodeEndpointResolver VirtualNodeEndpointResolver,
-	cloudMapInstanceReconciler CloudMapInstanceReconciler,
-	accountID string,
+	instancesReconciler InstancesReconciler,
 	log logr.Logger) ResourceManager {
 
-	return &cloudMapResourceManager{
+	return &defaultResourceManager{
 		k8sClient:                   k8sClient,
 		cloudMapSDK:                 cloudMapSDK,
 		virtualNodeEndpointResolver: virtualNodeEndpointResolver,
-		cloudMapInstanceReconciler:  cloudMapInstanceReconciler,
-		namespaceIDCache:            cache.NewLRUExpireCache(cloudMapNamespaceCacheMaxSize),
-		serviceIDCache:              cache.NewLRUExpireCache(cloudMapServiceCacheMaxSize),
-		accountID:                   accountID,
-		log:                         log,
+		instancesReconciler:         instancesReconciler,
+
+		namespaceSummaryCache: cache.NewLRUExpireCache(defaultNamespaceCacheMaxSize),
+		serviceSummaryCache:   cache.NewLRUExpireCache(defaultServiceCacheMaxSize),
+		log:                   log,
 	}
 }
 
-// cloudMapResourceManager implements ResourceManager
-type cloudMapResourceManager struct {
+// defaultResourceManager implements ResourceManager
+type defaultResourceManager struct {
 	k8sClient                   client.Client
 	cloudMapSDK                 services.CloudMap
 	virtualNodeEndpointResolver VirtualNodeEndpointResolver
-	cloudMapInstanceReconciler  CloudMapInstanceReconciler
-	namespaceIDCache            *cache.LRUExpireCache
-	serviceIDCache              *cache.LRUExpireCache
+	instancesReconciler         InstancesReconciler
 	accountID                   string
-	log                         logr.Logger
-}
 
-type cloudMapNamespaceSummary struct {
-	NamespaceID   string
-	NamespaceType string
+	namespaceSummaryCache *cache.LRUExpireCache
+	serviceSummaryCache   *cache.LRUExpireCache
+	log                   logr.Logger
 }
 
-type cloudMapServiceSummary struct {
-	NamespaceID string
-	ServiceID   string
+type serviceSummary struct {
+	serviceID               string
+	healthCheckCustomConfig *servicediscovery.HealthCheckCustomConfig
 }
 
-const (
-	cloudMapNamespaceCacheMaxSize = 100
-	cloudMapNamespaceCacheTTL     = 2 * time.Minute
-	cloudMapServiceCacheMaxSize   = 1024
-	cloudMapServiceCacheTTL       = 2 * time.Minute
-)
+func (m *defaultResourceManager) Reconcile(ctx context.Context, vn *appmesh.VirtualNode) error {
+	cloudMapConfig := vn.Spec.ServiceDiscovery.AWSCloudMap
+	nsSummary, err := m.findCloudMapNamespace(ctx, cloudMapConfig.NamespaceName)
+	if err != nil {
+		return err
+	}
+	if nsSummary == nil {
+		return fmt.Errorf("cloudMap namespace not found: %v", cloudMapConfig.NamespaceName)
+	}
+	svcSummary, err := m.findCloudMapService(ctx, nsSummary, cloudMapConfig.ServiceName)
+	if err != nil {
```

Comment:
err is nil even when service is missing? We might want to handle that error and proceed to service creation...

Reply:
yes :D, it's a valid case when deleting a virtualnode. (so the service may not have been created).

I used the convention like  if it's expected that something may not found, use a function like "findXXX", and return `nil,nil` when not found.